files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,155 @@
+# !/usr/bin/env python
+'''
+This code is for a specific daemon, process-docker.
+Which sends process and docker CPU/memory utilization data to DB every 2 mins.
+'''
+
+import sys, errno
+import os
+import time
+import syslog
+import signal
+import select
+import subprocess
+import re
+import swsssdk 
+import collections
+from datetime import datetime
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "process-docker"
+
+REDIS_HOSTIP = "127.0.0.1"
+
+# ========================== Syslog wrappers ==========================
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+# ========================== ProcessDocker class ==========================
+class ProcDockerStats:
+
+    def __init__(self):
+        self.swid = 0
+        self.running = False
+        self.handle = None
+        
+    def initialize(self):
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB")
+        # wipe out all data before updating with new values
+        self.state_db.delete_all_by_pattern('STATE_DB', 'Docker_Stats|*')
```

Comment:
This function is redundant with the constructor. Suggest moving the first two lines into the constructor, since they only need to be called once. The `delete_all_by_pattern` line should be moved into the main daemon loop.

Reply:
merged to constructor and moved delete_all_by_pattern in main loop

Reply:
Not moved?

Reply:
delete_all_pattern is part of the main loop now. It is getting called just before we start updating data. Am I missing something?

Reply:
I think you wound up reverting it back.

Reply:
Ah, now I see that you moved the call to `delete_all_by_pattern()` into the `update_dockerstats_command()` function, which is part of the main loop.